main-pipelines.py

```diff
@@ -1,51 +1,34 @@
-import time
-from functools import reduce
-from typing import List
-
 import ray
+import time
 from ray.data.dataset_pipeline import DatasetPipeline
-from whylogs.core.datasetprofile import DatasetProfile
+from whylogs.core import DatasetProfile, DatasetProfileView
 
-data_files = ["data/data1.csv", "data/data2.csv", "data/data3.csv"]
-data_files = ["data/short-data.csv"]
-
-
-def timer(name):
-    def wrapped(fn):
-        def timerfn():
-            print(f"========== {name} =============")
-            serial_start = time.time()
-            fn()
-            print(f"time {time.time() - serial_start} seconds")
-            print()
-        return timerfn
-    return wrapped
+from util import merge_profiles, timer, data_files
 
 
 @ray.remote
 class RemotePipelineActor:
     def __init__(self, pipeline: DatasetPipeline) -> None:
         self.pipeline = pipeline
 
-    def log_from_pipeline(self) -> DatasetProfile:
-        profile = DatasetProfile("")
+    def log_from_pipeline(self) -> DatasetProfileView:
+        profile = DatasetProfile()
+        # TODO jamie looks like splitting the files up still makes tracking a lot faster
         for df in self.pipeline.iter_batches(batch_size=10000, batch_format="pandas"):
-            profile.track_dataframe(df)
-        return profile
+            pass
+            profile.track(pandas=df)
+        return profile.view()
```

Comment:
@jamie256 Should I expect logging one large batch to be faster or slower than logging multiple smaller batches? I get much better perf by setting the `batch_size` to `10_000` from `1000` but I don't know if its a ray thing or a whylogs thing.

Reply:
whylogs v1 should be faster when logging more than 1000 rows, we will batch and pass arrays of data to the data sketches which handles larger batches more efficiently than single values or smaller batches.